### PR TITLE
chore: fix npm audit woes

### PR DESCRIPTION
**Pull request by akx**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Azure Pipelines' audit step has been broken for a while. 

This PR relaxes the `security` step:

* only check production dependencies
 * don't have low-level vulnerabilities break the build
* output parseable format to avoid huge ANSI art log

### Breaking Changes

None.

### CODE REVIEW

**Comment by anikethsaha** [comment]
Can you confirm whether its similar to this one or not ?

**Comment by akx** [comment]
@anikethsaha Not exactly. The main upgrade here is `jest`... however it looks like Jest 25.x requires native `async` support, which breaks builds on Node 6.

There are a couple of options I see here (cc @evilebottnawi):

* Drop support/testing for Node 6. 
  * It has been EOL since April 2019 or so, so I think we have no reason to support it anymore either.
  * Current versions of the plugin would continue to work, and future versions _might_ work.
* Change or remove the CI `npm audit` step. Maybe `--audit-level=moderate`?

**Comment by akx** [comment]
Should be fixed now... 

Turns out `audit --fix` has an `--only=prod` switch and `audit` without `--fix` has a `--production` switch. 🙄

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "prebuild": "npm run clean",
     "build": "cross-env NODE_ENV=production babel src -d dist --copy-files",
     "commitlint": "commitlint --from=master",
-    "security": "npm audit",
+    "security": "npm audit --audit-level=moderate --parseable --production",
```
Comment (evilebottnawi) [inline package.json]: Remove all flags

Comment (akx) [reply package.json]: Can you explain? That'll make this patch a full no-op and audit will be broken still.

Comment (evilebottnawi) [reply package.json]: Acorn release new version with fixed, we don't need it anymore, just update lock file

Comment (akx) [reply package.json]: That wouldn't seem to help due to how things are pinned in Jest's JSDOM:

```
$ rm package-lock.json

$ npm i
added 10 packages from 84 contributors, removed 619 packages, updated 55 packages and audited 894519 packages in 18.582s
found 7 moderate severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details

$ npm audit; npm audit --parseable
=== npm audit security report ===

# Run  npm install --save-dev jest@25.1.0  to resolve 7 vulnerabilities
SEMVER WARNING: Recommended action is a potentially breaking change

[... snip ...]

found 7 moderate severity vulnerabilities in 894519 scanned packages
  7 vulnerabilities require semver-major dependency updates.


install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>@jest/reporters>jest-runtime>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>jest-runner>jest-jasmine2>jest-runtime>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>jest-runner>jest-runtime>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>jest-runtime>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>jest-runner>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>@jest/core>jest-config>jest-environment-jsdom>jsdom>acorn	Y
install	acorn	moderate	npm install --save-dev jest@25.1.0	Regular Expression Denial of Service	https://npmjs.com/advisories/1488	jest>jest-cli>jest-config>jest-environment-jsdom>jsdom>acorn	Y

$
```

Comment (akx) [reply package.json]: So @evilebottnawi, what should we do?

Comment (evilebottnawi) [reply package.json]: Ignore it, we can't do something here

**Comment by evilebottnawi** [comment]
Done in master, thanks for the PR

**Pull request closed without merging**